### PR TITLE
Cans-255 add middle name and suffix

### PR DESCRIPTION
**Pull request by jonbwalker**
## Description
[CANS-255](https://osi-cwds.atlassian.net/browse/CANS-255)

## Tests

- [ ] I used TDD to develop the feature
- [x] I have included unit tests
- [ ] I have included new acceptance tests or modified existing ones
- [ ] I have included other tests

## Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] I ran all my tests locally which were green.
- [x] My code adds no new issues to Code Climate
- [x] I ran all the linters for the project.
- [x] My code is in a stable state ready to be deployable, but not necessarily complete.
- [x] I promise on my honor as a CWDS developer to ensure I don't break the build (or bring donuts if I leave things broken), to check linters, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

File: app/javascript/Application/components/Client/Client.js [hunk 1]
```diff
@@ -105,11 +103,14 @@ class Client extends Component {
...
                       {this.renderClientData(childData.first_name, 'First Name')}
+                      {this.renderClientData(childData.middle_name, 'Middle Name')}
                       {this.renderClientData(childData.last_name, 'Last Name')}
+                      {this.renderClientData(childData.suffix, 'Suffix')}
                       {this.renderClientData(childData.dob, 'Birth Date')}
-                      {this.renderClientData(childData.case_id, 'Case Number')}
-                      {this.renderClientData(this.formatClientId(childData.external_id), 'Client Id')}
                       {this.renderClientData(childData.county.name, 'County')}
+                      {this.renderClientData(this.formatClientId(childData.external_id), 'Client Id', 6)}
+                      {this.renderClientData(undefined, undefined, 6)}
```
Comment (jonbwalker) [inline app/javascript/Application/components/Client/Client.js]: This was done to create an offset for the caseId. In the mockup you'll see that case Id has 2 empty column spaces before it on page. 

Adding a blank grid is currently the recommended way to create an offset with material-ui-next
https://github.com/mui-org/material-ui/issues/11251

File: app/javascript/Application/components/Client/ClientFormValidator.test.js [hunk 2]
```diff
@@ -15,17 +15,15 @@ describe('ClientFormValidator', () => {
...
-      expect(validate('last_name', 'Mike John')).toBe(true);
-      expect(validate('last_name', 'Mike  John')).toBe(false);
-      expect(validate('last_name', 'Mike-John')).toBe(true);
-      expect(validate('last_name', 'mike')).toBe(true);
-      expect(validate('last_name', 'Mike1')).toBe(false);
-      expect(validate('last_name', 'Mike-')).toBe(false);
-      expect(validate('last_name', '   ')).toBe(false);
-      expect(validate('last_name', 'Mike#')).toBe(false);
-      expect(validate('last_name', '@')).toBe(false);
+    it('middle_name', () => {
```
Comment (denys-davydov) [inline app/javascript/Application/components/Client/ClientFormValidator.test.js]: We still need tests for last_name field

Comment (jonbwalker) [reply app/javascript/Application/components/Client/ClientFormValidator.test.js]: I put them back ebcff00

**Comment by jonbwalker** [comment]
Retest this please

**Comment by jonbwalker** [comment]
Retest this please

**Comment by jonbwalker** [comment]
Retest this please

**Pull request merged**